airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -472,3 +528,207 @@ class Labels(GoogleAdsStream):
+            if not self.current_state(customer_id):
+                yield child_slice
+                continue
+
+            parent_slice["resource_type"] = self.resource_type
+            for parent_record in self.parent_stream.read_records(sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=parent_slice):
+                self._process_parent_record(parent_record, child_slice)
+
+            # yield child slice if any records where read
+            if child_slice["id_to_time"]:
+                yield child_slice
+
+    def parse_response(self, response: SearchPager, stream_slice: MutableMapping[str, Any] = None) -> Iterable[Mapping]:
+        # update records with time obtained from parent stream
+        for record in super().parse_response(response):
+            record[self.cursor_field] = stream_slice.get("id_to_time", dict()).get(record[self.primary_key[0]])
+            yield record
+
+    def _update_state(self):
+        # if parent stream was used - copy state from it, otherwise set default state
+        if self.parent_stream.state:
+            self._state = {self.parent_stream_name: self.parent_stream.state}
+        else:
+            # full refresh sync without parent stream
+            self._state = {
+                self.parent_stream_name: {
+                    self.parent_cursor_field: pendulum.today().start_of("day").format(self.parent_stream.cursor_time_format)
+                }
+            }
+
+    def _read_deleted_records(self, stream_slice: MutableMapping[str, Any] = None):
+        # yield deleted records with id and time when record was deleted
+        for deleted_record_id in stream_slice.get("deleted_ids", []):
+            yield {self.id_field: deleted_record_id, "deleted_at": stream_slice["id_to_time"].get(deleted_record_id)}
+
+    def read_records(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_slice: MutableMapping[str, Any] = None, **kwargs
+    ) -> Iterable[Mapping[str, Any]]:
+        """
+        This method is overridden to read records using parent stream
+        """
+        self.incremental_sieve_logger.bump()
+        self.incremental_sieve_logger.info(f"Started reading records for slice: {stream_slice}")
+
+        # if state is present read records by ids from slice otherwise full refresh sync
+        yield from super().read_records(sync_mode, stream_slice=stream_slice)
+
+        # yield deleted items
+        yield from self._read_deleted_records(stream_slice)
+
+        self._update_state()
+
+    def get_query(self, stream_slice: Mapping[str, Any] = None) -> str:
+        query = GoogleAds.convert_schema_into_query(
+            schema=self.get_json_schema(), report_name=self.name, id_field=self.id_field, id_list=stream_slice.get("updated_ids")
+        )
+        return query
+
+
+class AdGroupCriterions(IncrementalEventsStream):
```

Comment:
we should be consistent with plural vs. singular in naming, this should be called `*Criterion` not `*Criterions`

Reply:
This is the old name. To avoid blocking this PR and certification, I created [new issue](https://github.com/airbytehq/airbyte/issues/30562) for this change as it is a breaking change.